arrow-flight/src/sql/client.rs

```diff
@@ -228,15 +228,18 @@ impl FlightSqlServiceClient<Channel> {
     }
 
     /// Execute a bulk ingest on the server and return the number of records added
-    pub async fn execute_ingest(
+    pub async fn execute_ingest<S>(
         &mut self,
         command: CommandStatementIngest,
-        batches: Vec<RecordBatch>,
-    ) -> Result<i64, ArrowError> {
+        stream: S,
```

Comment:
this is the key change -- to change the API to take a stream of record batches rather than a Vec of batches. This change I think allows the whole API to be streaming